tests/test_metrics.py

```diff
@@ -95,8 +95,126 @@ def test_mark(self):
             k=test_k
         )
 
-        # THEN the mean absolute recall @ k should equal the expected value
-        self.assertEqual(mean_abs_recall_k, 0.25)
+
+    def test_mapk(self):
+        """
+        Test mean absolute precision @ k (MAPK) function
+        """
+
+        # GIVEN test MAP@K metrics
+        test_actual = [['A', 'B', 'X'], ['A', 'B', 'Y']]
+        test_predicted = [['X', 'Y', 'Z'], ['A', 'Y', 'Z']]
+        test_k = 5
+
+        # WHEN metrics.mapk is run
+        mean_abs_recall_k = metrics.mapk(
+            actual=test_actual,
+            predicted=test_predicted,
+            k=test_k
+        )
+
+        # THEN the mean absolute precision @ k should the average
+        #  precision over the two sets of predictions
+        self.assertEqual(mean_abs_recall_k, ((1/9) + (2/9)) / 2)
+
+    def test_apk(self):
```

Comment:
It would be great if someone could check the numbers in these tests to make sure the math is actually correct :)